packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -139,11 +179,10 @@ exports.sourceNodes = async (
   // Update syncToken
   const nextSyncToken = currentSyncData.nextSyncToken
 
-  // Store our sync state for the next sync.
-  const newState = {}
-  newState[createSyncToken()] = nextSyncToken
-  setPluginStatus(newState)
-
+  await Promise.all([
+    cache.set(CACHE_SYNC_KEY, currentSyncData),
```

Comment:
This doesn't seem to work. The cache is not set, and IMO probably because currentSyncData is not string-safe, or a plain object?

Reply:
Yep, error message on setting the cache is `Converting circular structure to JSON`

Reply:
I guess we have to store the data in a raw state as these won't contain any circular references.